pkg/operator/ceph/file/subvolumegroup/controller.go

```diff
@@ -187,6 +187,25 @@ func (r *ReconcileCephFilesystemSubVolumeGroup) reconcile(request reconcile.Requ
 		if cephCluster.Spec.External.Enable {
 			logger.Warningf("external subvolume group %q deletion is not supported, delete it manually", namespacedName)
 		} else {
+			if opcontroller.IsCleanupRequired(cephFilesystemSubVolumeGroup.GetAnnotations()) {
+				logger.Infof("starting cleanup of the ceph resources for subVolumeGroup %q", namespacedName)
+				cleanupConfig := map[string]string{
+					opcontroller.CephFSSubVolumeGroupNameEnv: cephFilesystemSubVolumeGroup.Spec.Name,
+					opcontroller.CephFSVolumeNameEnv:         cephFilesystemSubVolumeGroup.Spec.FilesystemName,
+					opcontroller.CephFSNamesaceEnv:           "csi", // TODO: is it always "csi" for cephFS
+					opcontroller.CephFSMetaDataPoolNameEnv:   fmt.Sprintf("%s-%s", cephFilesystemSubVolumeGroup.Spec.FilesystemName, "metadata"),
+				}
+				cleanup := opcontroller.NewResourceCleanup(cephFilesystemSubVolumeGroup, &cephCluster, r.opConfig.Image, cleanupConfig)
+				success, err := cleanup.Start(r.clusterInfo.Context, r.context.Client)
+				if err != nil {
+					return opcontroller.ImmediateRetryResult, errors.Wrapf(err, "failed to clean up filesystem subvolume group %q", cephFilesystemSubVolumeGroup.Name)
+				}
+				if !success {
+					return opcontroller.ImmediateRetryResult, errors.Wrapf(err, "Waiting for ceph cleanup job to complete successfully for cephFilesystemSubVolumeGroup %q", cephFilesystemSubVolumeGroup.Name)
+				}
+				logger.Infof("successfully cleaned up all the ceph resources created by the cephFilesystemSubVolumeGroup %q", cephFilesystemSubVolumeGroup.Name)
```

Comment:
If we get here it just means the job was successfully started. We don't know if it successfully completed, so we should always requeue the event, right?